solidity/random-beacon/README.adoc

```diff
@@ -1 +1,296 @@
-= Keep Random Beacon
\ No newline at end of file
+= Keep Random Beacon v2
+
+The Keep Network requires a trusted source of randomness for the process of
+trustless group election. While the network requires that randomness to function
+correctly, the source of randomness is itself broadly applicable. This trusted
+source of randomness takes the form of a BLS Threshold Relay.
+
+== Overview
+
+The threshold relay is a way of generating verifiable randomness that is
+resistant to bad actors both in the relay network and on the anchoring Ethereum
+blockchain. The basic functioning of the relay is:
+
+- Some number of groups exist in the relay.
+- An arbitrary seed value `v_s` counts as the first entry in the relay.
+- A request `r_i` is dispatched to the chain for a new entry.
+- The previous entry `v_s` is used to choose a group to produce the response to
+  the request.
+- `v_s` is signed by at least a subset of the chosen group members, and the
+  resulting signature is the entry generated in response to the request. It is
+  published to the anchoring blockchain as the entry `v_i`.
+- The new entry `v_i` may trigger the formation of a new group from the set of
+  all members in the relay.
+- A group expires after a certain amount of time.
+
+== Prior Work
+
+Smart contracts for the first version of the random beacon are available in
+link:./../../solidity-v1[`solidity-v1` directory]. The new version uses the same
```

Comment:
Maybe it will be better to link to `https://github.com/keep-network/keep-core/tree/main/solidity-v1` instead a relative path, as if we publish this README inside a NPM package the path won't resolve.

Reply:
Good point, will do!

Reply:
https://github.com/keep-network/keep-core/pull/2746/commits/80f16897fe14305ae1a2aaf52b5352694349c718